addons/stock/product.py

```diff
@@ -76,7 +76,8 @@ def _get_domain_locations(self, cr, uid, ids, context=None):
             if type(context['location']) == type(1):
                 location_ids = [context['location']]
             elif type(context['location']) in (type(''), type(u'')):
-                domain = [('complete_name','ilike',context['location'])]
+                location = context['location'].replace('/', ' / ')
+                domain = [('complete_name', 'ilike', location)]
```

Comment:
The name of context['location'] is the name returned in name_get function('WH/Stock'). The only way I've found is to add spaces between / and search by complete name('Physical locations / WH / Stock'). Search the name  with the text after / will return various locations (WH/Stock, Chic/Stock)